internal/scheduler/database/job_repository_test.go

```diff
@@ -545,6 +719,66 @@ func TestFetchJobRunLeases(t *testing.T) {
 	}
 }
 
+func TestFetchLatestSerials(t *testing.T) {
+	dbJobs, _ := createTestJobs(10)
+	dbRuns, _ := createTestRuns(10)
+
+	tests := map[string]struct {
+		dbJobs             []Job
+		dbRuns             []Run
+		latestJobSerial    int64
+		latestJobRunSerial int64
+	}{
+		"all jobs": {
+			dbJobs:          dbJobs,
+			latestJobSerial: 10,
+		},
+		"some jobs": {
+			dbJobs:          dbJobs[:5],
+			latestJobSerial: 5,
+		},
+		"all runs": {
+			dbRuns:             dbRuns,
+			latestJobRunSerial: 10,
+		},
+		"some runs": {
+			dbRuns:             dbRuns[:5],
+			latestJobRunSerial: 5,
+		},
+		"both jobs and runs": {
+			dbJobs:             dbJobs,
+			dbRuns:             dbRuns,
+			latestJobSerial:    10,
+			latestJobRunSerial: 10,
+		},
+		"empty db": {},
+	}
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			err := withJobRepository(func(repo *PostgresJobRepository) error {
+				ctx, cancel := armadacontext.WithTimeout(armadacontext.Background(), 5*time.Second)
+
+				// Set up db
+				err := database.UpsertWithTransaction(ctx, repo.db, "jobs", tc.dbJobs)
+				require.NoError(t, err)
+				err = database.UpsertWithTransaction(ctx, repo.db, "runs", tc.dbRuns)
+				require.NoError(t, err)
+
+				// Fetch updates
+				jobSerial, jobRunSerial, err := repo.FetchLatestSerials(ctx)
+				require.NoError(t, err)
+
+				// Assert results
+				assert.Equal(t, tc.latestJobSerial, jobSerial)
+				assert.Equal(t, tc.latestJobRunSerial, jobRunSerial)
+				cancel()
```

Comment:
Maybe it is better to do the `cancel` in a `t.Cleanup` to prevent context leaks, even though this are tests, no reason to have lingering goroutines as the `require.NoError` will terminate in case of an error and the context won't get cancelled :)

Reply:
I was following convention here but yeah I dunno why the code doesn't already do that